git_machete/tests/functional/test_machete.py

```diff
@@ -25,6 +28,170 @@ def get_head_commit_hash() -> str:
     return os.popen("git rev-parse HEAD").read().strip()
 
 
+class FakeCommandLineOptions:
+    opt_debug: bool = False
+    opt_verbose: bool = False
+
+    def __init__(self) -> None:
+        self.opt_as_root: bool = False
+        self.opt_branch: Optional[str] = None
+        self.opt_checked_out_since: Optional[str] = None
+        self.opt_color: str = "auto"
+        self.opt_down_fork_point: Optional[str] = None
+        self.opt_draft: bool = False
+        self.opt_fetch: bool = False
+        self.opt_fork_point: Optional[str] = None
+        self.opt_inferred: bool = False
+        self.opt_list_commits: bool = False
+        self.opt_list_commits_with_hashes: bool = False
+        self.opt_merge: bool = False
+        self.opt_n: bool = False
+        self.opt_no_detect_squash_merges: bool = False
+        self.opt_no_edit_merge: bool = False
+        self.opt_no_interactive_rebase: bool = True
+        self.opt_onto: Optional[str] = None
+        self.opt_override_to: Optional[str] = None
+        self.opt_override_to_inferred: bool = False
+        self.opt_override_to_parent: bool = False
+        self.opt_push_tracked: Optional[bool] = True
+        self.opt_push_untracked: Optional[bool] = True
+        self.opt_return_to: str = "stay"
+        self.opt_roots: List[str] = list()
+        self.opt_start_from: str = "here"
+        self.opt_stat: bool = False
+        self.opt_sync_github_prs: bool = False
+        self.opt_unset_override: bool = False
+        self.opt_yes: bool = True
```

Comment:
This is needed to set `opt_yes` and `opt_no_interactive_rebase` to true, in order to perform `__sync_before_creating_pr`. The other way to acomplish such a settitng in tests, is to call `create_github_pr` directly from MacheteClient class, like it is in lines from 1780 till bottom. Which way would be better?

Reply:
Other option would be to inherit from CommandLineOptions and in init first call init from super and then adjusting only these variables that should be adjusted (preferable with some comment justifying these adjustment to next generation)?

Reply:
Sounds good, this way we can clearly see which options are actually overrridden